src/test/regression/test_exit_after_signal_sched.rs

```diff
@@ -0,0 +1,46 @@
+/// Regression test for [#1623](https://github.com/shadow/shadow/issues/1623)
+///
+/// Only designed to run under Shadow, as it relies on Shadow's deterministic scheduling.
+fn main() {
+    let (read_fd, write_fd) = nix::unistd::pipe2(nix::fcntl::OFlag::O_NONBLOCK).unwrap();
+
+    // Write to the pipe until its buffer is full.
+    loop {
+        let buffer = [0; 1024];
+        match nix::unistd::write(write_fd, &buffer) {
+            Ok(_) => (),
+            Err(nix::errno::Errno::EWOULDBLOCK) => break,
+            Err(e) => panic!("Unexpected result: {}", e),
+        }
+    }
+
+    // Clear O_NONBLOCK, making subsequent write operations blocking.
+    nix::fcntl::fcntl(
+        write_fd,
+        nix::fcntl::FcntlArg::F_SETFL(nix::fcntl::OFlag::empty()),
+    )
+    .unwrap();
+
+    let child_finished = std::sync::Arc::new(std::sync::atomic::AtomicBool::new(false));
+    {
+        let child_finished = child_finished.clone();
+        std::thread::spawn(move || {
+            // This should initially block.
+            assert_eq!(1, nix::unistd::write(write_fd, &[0]).unwrap());
+            // Signal to the parent that the `write` returned.
+            child_finished.store(true, std::sync::atomic::Ordering::SeqCst);
+        })
+    };
```

Comment:
The `child_finished` doesn't seem to be read anywhere?

Reply:
Fixed; thanks!